src/Workspaces/Remote/Core/ExportProviderBuilder.cs

```diff
@@ -182,10 +183,14 @@ protected virtual void PerformCacheDirectoryCleanup(DirectoryInfo directoryInfo,
 
     protected abstract bool ContainsUnexpectedErrors(IEnumerable<string> erroredParts);
 
-    private void ReportCompositionErrors(CompositionConfiguration configuration, ComposableCatalog catalog)
+    /// <returns>True if there was an unexpected composition error, false otherwise.</returns>
+    private bool CheckForAndReportCompositionErrors(CompositionConfiguration configuration, ComposableCatalog catalog)
     {
+        var hasErrors = false;
+
         foreach (var exception in catalog.DiscoveredParts.DiscoveryErrors)
         {
+            hasErrors = true;
             LogError($"Encountered exception in the MEF composition: {exception.Message}");
```

Comment:
Will this come back to us as telemetry?

Reply:
And yes, at least indirectly: this goes through the gold bar reporting, which then logs telemetry for this message. I can see telemetry for myself or @phil-allen-msft hitting this exact line.